app/models/voucher.rb

```diff
@@ -7,7 +7,8 @@ class Voucher < ApplicationRecord
 
   belongs_to :enterprise
 
-  has_many :adjustments, as: :originator, class_name: 'Spree::Adjustment'
+  has_many :adjustments, as: :originator, class_name: 'Spree::Adjustment', inverse_of: :voucher,
+           dependent: :nullify
```

Comment:
I raised the same thing three weeks ago:
https://github.com/openfoodfoundation/openfoodnetwork/pull/10587#discussion_r1170664585

I also think that the nullify option should be removed to make it impossible to delete vouchers from the database. But since we don't have any functionality doing it at the moment, it doesn't really matter. So I wouldn't stop this already huge PR because of this but it would be nice to clean up at some point.